src/hotspot/share/runtime/os.cpp

```diff
@@ -1002,8 +1002,14 @@ void os::print_date_and_time(outputStream *st, char* buf, size_t buflen) {
 
   struct tm tz;
   if (localtime_pd(&tloc, &tz) != NULL) {
-    ::strftime(buf, buflen, "%Z", &tz);
-    st->print("Time: %s %s", timestring, buf);
+    wchar_t w_buf[80];
+    size_t n = ::wcsftime(w_buf, 80, L"%Z", &tz);
```

Comment:
According to https://linux.die.net/man/3/wcsftime: "The Linux implementation of this interface may differ (consult the corresponding Linux manual page for details of Linux behavior), or the interface may not be implemented on Linux."

Also, the L"%Z" notation is Windows-specific.

Maybe we should use the new code only on Windows?

An alternative is to use the C++ standard library (std::wcsftime and std:: wcstombs). However, these part of the STL is not yet permitted -- see https://bugs.openjdk.java.net/browse/JDK-8208089

Reply:
After further investigation -- wcsftime is in C99. Also, we use it here only inside `if (localtime_pd(&tloc, &tz) != NULL)`. I supposed any Linux distro that has a minimal of locale support to make that function return non-null would have a working implementation of wcsftime.

So I think this code is OK. The only change I request is to change ` L"%Z"` to `"%Z"`

Reply:
Upon even more investigation, I was completely wrong :-)

According https://en.cppreference.com/w/cpp/language/string_literal, the "L" prefix is for `wchar_t` string literals. So your code is correct.

Reply:
@iklam Thanks for your review and making sure the change is ok.